crates/miden-testing/tests/scripts/faucet.rs

```diff
@@ -275,3 +289,153 @@ async fn prove_burning_fungible_asset_on_existing_faucet_succeeds() -> anyhow::R
+async fn test_public_note_creation_with_script_from_datastore() -> anyhow::Result<()> {
+    let mut builder = MockChain::builder();
+    let faucet = builder.add_existing_faucet(Auth::BasicAuth, "TST", 200, None)?;
+
+    // Parameters for the PUBLIC P2ID note that will be created by the faucet
+    let recipient_account_id = AccountId::try_from(ACCOUNT_ID_PRIVATE_SENDER)?;
+    let amount = Felt::new(75);
+    let tag = NoteTag::for_public_use_case(0, 0, miden_objects::note::NoteExecutionMode::Local)?;
+    let aux = Felt::new(27);
+    let note_execution_hint = NoteExecutionHint::on_block_slot(5, 6, 7);
+    let note_type = NoteType::Public; // PUBLIC note
+
+    // Create the expected P2ID note using the helper function for debugging purposes
+    let mut rng = RpoRandomCoin::new([Felt::from(0u32); 4].into());
+    let expected_p2id_note = create_p2id_note(
+        faucet.id(),
+        recipient_account_id,
+        vec![FungibleAsset::new(faucet.id(), amount.into())?.into()],
+        note_type,
+        aux,
+        &mut rng,
+    )?;
+
+    // Extract recipient information from the expected note
+    let p2id_recipient = expected_p2id_note.recipient();
+    let p2id_script_root = p2id_recipient.script().root();
+    let serial_num = p2id_recipient.serial_num();
+    let target_account_suffix = recipient_account_id.suffix();
+    let target_account_prefix = recipient_account_id.prefix().as_felt();
+
+    println!("SERIAL_NUM: {:?}", serial_num);
+    println!("SCRIPT_ROOT: {:?}", p2id_script_root);
+    println!("INPUTS_COMMITMENT: {:?}", expected_p2id_note.inputs().commitment());
+    println!("RECIPIENT {:?}", p2id_recipient.digest());
+
+    let note_script_code = format!(
+        "
+            use.miden::note
+            
+            begin
+                # Build recipient hash from SERIAL_NUM, SCRIPT_ROOT, and INPUTS_COMMITMENT
+                push.{script_root}
+                # => [SCRIPT_ROOT]
+                
+                push.{serial_num}
+                # => [SERIAL_NUM, SCRIPT_ROOT]
+
+                # Store P2ID inputs (target account ID) in memory at address 0
+                # P2ID inputs are: [target.suffix(), target.prefix()]
+                push.{target_suffix}.{target_prefix} push.0.0
+                push.0 mem_storew dropw
+                # Memory[0] = [0, 0, target_prefix, target_suffix]
+
+                push.2 push.0
+                # => [inputs_ptr, num_inputs, SERIAL_NUM, SCRIPT_ROOT]
+
+                exec.note::build_recipient
+                # => [RECIPIENT]
+
+                # Now call distribute with the computed recipient
+                push.{note_execution_hint}
+                push.{note_type}
+                push.{aux}
+                push.{tag}
+                push.{amount}
+                # => [amount, tag, aux, note_type, execution_hint, RECIPIENT]
+
+                call.::miden::contracts::faucets::basic_fungible::distribute
+                # => [note_idx, pad(15)]
```

Comment:
Is it essential for the test to call `distribute` instead of `output_note::create`?